### PR TITLE
Update release guide 

### PR DESCRIPTION
**Pull request by georgysavva**

### CODE REVIEW

File: documentation/docs/teamresources/release.md [hunk 1]
```diff
@@ -2,11 +2,7 @@
 
 1. Create a PR into `develop` updating the banner version (`plugins/banner.AppVersion`) and mentioning the changes in `CHANGELOG.md`
 2. Create a PR merging `develop` into `master`
-3. Create a release via the release page with the same changelog entries as in `CHANGELOG.md` for the given version tagging the `master` branch
-4. Pray that the CI gods let the build pass
+3. Go to release workflow https://github.com/iotaledger/goshimmer/actions/workflows/release.yml and click the gray "Run workflow" button to configure the release process.
+4. In "Branch" field set `master`, in "Tag name" set current version, in "Release description" paste the changes recently added to `CHANGELOG.md`. Click the green "Run workflow" to trigger the automatic release and deployment process.
 5. Check that the binaries are working
```
Comment (apenzk) [inline documentation/docs/teamresources/release.md:7]:
```suggestion
5. Check that the binaries are working.
```

File: documentation/docs/teamresources/release.md [hunk 2]
```diff
@@ -2,11 +2,7 @@
 
 1. Create a PR into `develop` updating the banner version (`plugins/banner.AppVersion`) and mentioning the changes in `CHANGELOG.md`
 2. Create a PR merging `develop` into `master`
```
Comment (apenzk) [inline documentation/docs/teamresources/release.md:4]:
```suggestion
2. Create a PR merging `develop` into `master`.
```

File: documentation/docs/teamresources/release.md [hunk 3]
```diff
@@ -2,11 +2,7 @@
 
 1. Create a PR into `develop` updating the banner version (`plugins/banner.AppVersion`) and mentioning the changes in `CHANGELOG.md`
```
Comment (apenzk) [inline documentation/docs/teamresources/release.md:3]:
```suggestion
1. Create a PR into `develop` updating the banner version (`plugins/banner.AppVersion`) and mentioning the changes in `CHANGELOG.md`.
```

**Review by apenzk — APPROVED** [review]

**Comment by georgysavva** [comment]
@apenzk thank you for the fixes!

**Pull request merged**